src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -47,11 +47,6 @@ internal SyntaxTreeSemanticModel(CSharpCompilation compilation, SyntaxTree synta
             _syntaxTree = syntaxTree;
             _ignoresAccessibility = ignoreAccessibility;
 
-            if (!this.Compilation.SyntaxTrees.Contains(syntaxTree))
```

Comment:
I took the approach of removing this entirely, as on every path except through the AnalyzerDriver we will hit https://github.com/dotnet/roslyn/blob/main/src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs#L2248-L2251, and `GetBinderFactory` below will assert if the semantic model and syntax tree don't line up. If we think that peppering the AnalyzerDriver with a few of those checks is not good, though, we could add this check back in, but using the `ContainsSyntaxTree` check on `CSharpCompilation` instead, which is `O(1)` rather than `O(n)`.